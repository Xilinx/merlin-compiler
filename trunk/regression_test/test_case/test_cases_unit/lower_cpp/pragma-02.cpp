//(C) Copyright 2016-2021 Xilinx, Inc.
//All Rights Reserved.
//
//Licensed to the Apache Software Foundation (ASF) under one
//or more contributor license agreements.  See the NOTICE file
//distributed with this work for additional information
//regarding copyright ownership.  The ASF licenses this file
//to you under the Apache License, Version 2.0 (the
//"License"); you may not use this file except in compliance
//with the License.  You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
//Unless required by applicable law or agreed to in writing,
//software distributed under the License is distributed on an
//"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
//KIND, either express or implied.  See the License for the
//specific language governing permissions and limitations
//under the License. (edited)
namespace ns {
  const int FC = 10;
  enum {DIM=1};
class A {
  int a[10];
  public:
  A() {
    // the following 'a' should be updated into "this_->a"
#pragma HLS array_patition variable=a factor=FC dim=DIM
    a[0] = 1;
  }
  int get(int i) {
    return a[i];
  }
  void set(int i, int val) {
    a[i] = val;
  }
};
}
enum _ram_type {
  RAM_1P_BRAM = 0,
};
#pragma ACCEL kernel
void top(int *a) {
// do not replace enumerator if its declaration is in the global scope
#pragma HLS interface variable=a core=RAM_1P_BRAM
  ns::A tmp;
  for (int i = 0; i < 10; ++i) {
    tmp.set(i, a[i]);
  }
}
