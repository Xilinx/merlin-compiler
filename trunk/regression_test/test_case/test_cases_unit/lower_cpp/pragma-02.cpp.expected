
struct ::ns_A 
{
  int a[10];
}
;
const int _ZN2ns2FCE = 10;

void p_ZN2ns1AC2Ev(struct ns_A *this_)
{
// the following 'a' should be updated into "this_->a"
  
#pragma HLS array_patition variable=this_->a factor=_ZN2ns2FCE dim=1
  this_ -> a[0] = 1;
}

int p_ZN2ns1A3getEi(struct ns_A *this_,int i)
{
  return this_ -> a[i];
}

void p_ZN2ns1A3setEii(struct ns_A *this_,int i,int val)
{
  this_ -> a[i] = val;
}
namespace ns
{
static void __merlin_dummy_SgVariableDeclaration_const_int_();
static void __merlin_dummy_SgEnumDeclaration_enum_ano_();
static void __merlin_dummy_SgClassDeclaration_struct_A_();
}
enum _ram_type {RAM_1P_BRAM=0} ;
#pragma ACCEL kernel

void top(int *a)
{
// do not replace enumerator if its declaration is in the global scope
#pragma HLS interface variable=a core=RAM_1P_BRAM
struct ns_A tmp;
p_ZN2ns1AC2Ev(&tmp);
;
for (int i = 0; i < 10; ++i) {
p_ZN2ns1A3setEii(&tmp,i,a[i]);
}
}
